app/views/shared/forms/_date_input_fields.html.erb

```diff
@@ -11,8 +11,8 @@
             <%= content_tag(:span, t(".#{field_name}_hint"), default: '', id: "#{field_name}-hint".dasherize, class: ['govuk-hint']) %>
         <% end %>
 
-        <% form.object.errors[field_name].each do |message| %>
-            <%= content_tag(:span, message, id: "#{field_name}-error".dasherize, class: 'govuk-error-message') if form.object.errors[field_name].any? %>
+        <% if form.object.errors[field_name].any? %>
+          <%= content_tag(:span, form.object.errors[field_name].first, id: "#{field_name}-error".dasherize, class: ['govuk-error-message']) %>
```

Comment:
The `if form.object.errors[field_name].any?` in the original wasn't doing anything as `orm.object.errors[field_name].each` would only pass something into `message` and do the iteration if there was something present.

This code fixes that, but also only shows the message for the first error. If that was the desired result then fine. Otherwise, just remove the `if form.object.errors[field_name].any?` from the original.

Reply:
only showing the first message is the intention here.